drivers/flash/flash_stm32g4x.c

```diff
@@ -159,6 +182,8 @@ static int erase_page(const struct device *dev, unsigned int offset)
 	/* Wait for the BSY bit */
 	rc = flash_stm32_wait_flash_idle(dev);
 
+	flush_cache(regs);
```

Comment:
According to STM32G4 Reference Manual, and I found that:
Chapter 3.3.7

> **Programming and caches**
> If a Flash memory write access concerns some data in the data cache, the Flash write
> access modifies the data in the Flash memory and the data in the cache.
> If an erase operation in Flash memory also concerns data in the data or instruction cache,
> you have to make sure that these data are rewritten before they are accessed during code
> execution. If this cannot be done safely, it is recommended to flush the caches

So flush is necessary for erase operation only.

Reply:
Thanks. Fixed.

Reply:
Is it ?

Reply:
> Therefore I would flush cashes only prior the erase operation.

Reply:
Oups, got errored by github.